src/Symfony/Bundle/SecurityBundle/Resources/config/templating_twig.php

```diff
@@ -13,12 +13,13 @@
 
 use Symfony\Bridge\Twig\Extension\LogoutUrlExtension;
 use Symfony\Bridge\Twig\Extension\SecurityExtension;
+use Symfony\Component\Security\Http\Logout\LogoutUrlGeneratorInterface;
 
 return static function (ContainerConfigurator $container) {
     $container->services()
         ->set('twig.extension.logout_url', LogoutUrlExtension::class)
             ->args([
-                service('security.logout_url_generator'),
+                service(LogoutUrlGeneratorInterface::class),
```

Comment:
why change? we don't reference autowiring aliases usually

Reply:
It allows to change the implementation of the interface without replacing the service. `security.logout_url_generator` must be an instance of the class `LogoutUrlGenerator` for the `FirewallListener`. Same for the data collector service.

I don't have an actual use-case in mind.

Reply:
this should be reverted, we don't do this usually